staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go

```diff
@@ -92,18 +93,174 @@ const (
 	defaultPodLogsTimeout = 20 * time.Second
 )
 
+var defaultContainerNameFromRefSpecRegexp = regexp.MustCompile(`spec\.(?:initContainers|containers|ephemeralContainers){(.+)}`)
+
+// LogsFlags reflects the information that the CLI is gathering via flags.
+// They will be converted to PodLogOptions for runtime requirements.
+type LogsFlags struct {
+	// required objects
+	RESTClientGetter genericclioptions.RESTClientGetter
+	ResourceBuilder  *resource.Builder
```

Comment:
`ResourceBuilderFlags` is preferred rather than this, see https://github.com/kubernetes/kubernetes/blob/c75d254beb662ea73736d2f123d0382b7f91dfbd/staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go#L88